src/lua/api-gateway/util/OauthClient.lua

```diff
@@ -42,12 +75,21 @@ function OauthClient:makeProfileCall(internalPath, oauth_host)
 
     oauth_host = oauth_host or ngx.var.oauth_host
     ngx.log(ngx.INFO, "profileCall request to host=", oauth_host)
+    local startTime = os.clock()
     local res = ngx.location.capture(internalPath, { share_all_vars = true })
+    local endTime = os.clock()
+    self:increment(self.oauthHttpCallsMetric, 1)
+
+    local elapsedTime = os.difftime(endTime,startTime) * 1000
+    local elapsedTimeMetric = self.oauthHttpCallsMetric '.makeProfileCall.duration'
```

Comment:
Does this code actually work? How the concatenation is done? This is exactly the reason why i suggested a function that compresses all these metric name composings.